tools/lldb/dylan/commands.py

```diff
@@ -47,44 +52,52 @@ def dylan_bt(debugger, command, result, internal_dict):
       fmt += ' {mod}'
     if file_name:
       fmt += ' at {file}:{line}'
-    print fmt.format(
+    print (fmt.format(
       num=frame_idx,
       func=function_name,
       mod=module,
       addr=address,
       file=file_name,
       line=line_number
-      )
+      ))
 
 def dylan_break_gf(debugger, command, result, internal_dict):
   command_args = shlex.split(command)
-  usage = 'usage: %prog [gf]'
-  description = 'Set a breakpoint that covers every method on the generic function.'
-  parser = optparse.OptionParser(description=description, prog='dylan-break-gf', usage=usage)
-
+  description = "Set a breakpoint that covers every method on the generic function."
+  epilog="""
+It is not possible to set GF breakpoints in libraries that haven't
+been loaded yet, so it may be necessary to run the debuggee before
+using this command
+"""
+  parser = argparse.ArgumentParser(description=description, prog='dylan-break-gf', epilog=epilog)
+  parser.add_argument('gf', nargs='+', help='binding name in the form "name:module:library"')
   try:
-    (options, args) = parser.parse_args(command_args)
+    options = parser.parse_args(command_args)
   except:
     return
 
   target = debugger.GetSelectedTarget()
-  for arg in args:
+  for arg in options.gf:
     try:
       (binding, module, library) = utils.parse_binding(arg)
       symbol_name = mangling.dylan_mangle_binding(binding, module, library)
     except utils.InvalidBindingIdentifier:
-      print 'Invalid binding name: %s' % arg
+      print('Invalid binding name: %s' % (arg,))
       continue
-    expression = '(dylan_value)&%s' % symbol_name
+    expression = '(dylan_value)&%s' % (symbol_name,)
     gf = target.CreateValueFromExpression(symbol_name, expression)
     if gf.GetError().Fail():
-      print "No generic function %s was found." % arg
+      print("No generic function %s was found." % (arg,))
       continue
-    methods = dylan_generic_function_methods(gf)
+    try:
+      methods = dylan_generic_function_methods(gf)
+    except:
```

Comment:
I would shy away from bare `except:` since it catches things like `Ctrl-c`. If it's not clear what exception is being thrown `except Exception:` should be preferred.

Reply:
I changed this and also the other places where it was written like this.